packages/openneuro-server/src/graphql/resolvers/user.ts

```diff
@@ -3,8 +3,20 @@
  */
 import User from "../../models/user"
 
+
+
 export const user = (obj, { id }) => {
-  return User.findOne({ id }).exec()
+  
+  function isValidOrcid(orcid: string): boolean {
+    return /^[0-9]{4}-[0-9]{4}-[0-9]{4}-[0-9]{3}[0-9X]$/.test(orcid || '');
+  }
+
+  if (isValidOrcid(id)) {
+    return User.findOne({ "orchid": id }).exec()
```

Comment:
```suggestion
    return User.findOne({ "orcid": id }).exec()
```

Reply:
One typo here but another limitation is there's two kinds of user records here:

```json
{
  "provider": "orcid",
  "providerId": "0000-0000..."
}
```

```json
{
  "provider": "google",
  "orcid": "0000-0000..."
}
```

This branch needs to find both of these. Once we've migrated Google users to ORCID logins, most users will fall into the first group that won't match here.

Reply:
updated typo.

Reply:
@nellh something like this?

export const user = (obj, { id, key }) => {
  function isValidOrcid(orcid: string): boolean {
    return /^[0-9]{4}-[0-9]{4}-[0-9]{4}-[0-9]{3}[0-9X]$/.test(orcid || "")
  }

  if (isValidOrcid(id)) {
    if (key === 'orcid') {
      return User.findOne({ "orcid": id }).exec(); // Search by orcid if key is 'orcid'
    } else {
      return User.findOne({ "providerId": id }).exec(); // Otherwise, search by providerId
    }
  } else {
    return User.findOne({ "id": id }).exec(); // Search by id if not a valid ORCID
  }
}

Reply:
It should be a single query, you can use the `{$or: [{ "orcid": id }, {"providerId": id}]}` operator to check if the id is in either field. No need for a key argument.

Reply:
ty - that was helpful. I have that added now